tool-openssl/pkcs8.cc

```diff
@@ -84,141 +77,59 @@ static const std::unordered_set<std::string> kSupportedPRFs = {
-            return false;
-        }
-        source = password;
-        return true;
-    }
-
-    if (source.compare(0, 5, "file:") == 0) {
-        std::string path = source.substr(5);
-        bssl::UniquePtr<BIO> bio(BIO_new_file(path.c_str(), "r"));
-        if (!bio) {
-            fprintf(stderr, "Cannot open password file\n");
-            return false;
-        }
-        char buf[DEFAULT_MAX_SENSITIVE_STRING_LENGTH] = {};
-        int len = BIO_gets(bio.get(), buf, sizeof(buf));
-        if (len <= 0) {
-            OPENSSL_cleanse(buf, sizeof(buf));
-            fprintf(stderr, "Cannot read password file\n");
-            return false;
-        }
-        const bool possible_truncation = (static_cast<size_t>(len) == DEFAULT_MAX_SENSITIVE_STRING_LENGTH - 1 && 
-                                        buf[len - 1] != '\n' && buf[len - 1] != '\r');
-        if (possible_truncation) {
-            OPENSSL_cleanse(buf, sizeof(buf));
-            fprintf(stderr, "Password file content too long\n");
-            return false;
-        }
-        size_t buf_len = len;
-        while (buf_len > 0 && (buf[buf_len-1] == '\n' || buf[buf_len-1] == '\r')) {
-            buf[--buf_len] = '\0';
-        }
-        source = buf; 
-        OPENSSL_cleanse(buf, sizeof(buf));
-        return true;
-    }
-
-    if (source.compare(0, 4, "env:") == 0) {
-        std::string env_var = source.substr(4);
-        if (env_var.empty()) {
-            fprintf(stderr, "Empty environment variable name\n");
-            return false;
-        }
-        const char* env_val = getenv(env_var.c_str());
-        if (!env_val) {
-            fprintf(stderr, "Environment variable '%s' not set\n", env_var.c_str());
-            return false;
-        }
-        size_t env_val_len = strlen(env_val);
-        if (env_val_len == 0) {
-            fprintf(stderr, "Environment variable '%s' is empty\n", env_var.c_str());
-            return false;
-        }
-        if (env_val_len > DEFAULT_MAX_SENSITIVE_STRING_LENGTH) {
-            fprintf(stderr, "Environment variable value too long\n");
-            return false;
-        }
-        source = std::string(env_val);
-        return true;
-    }
-    fprintf(stderr, "Invalid password format (use pass:, file:, or env:)\n");
+static bool validate_prf(const std::string &prf_name) {
+  if (kSupportedPRFs.find(prf_name) == kSupportedPRFs.end()) {
+    fprintf(stderr, "Only hmacWithSHA1 PRF is supported\n");
     return false;
+  }
+  return true;
 }
 
-// Custom deleter for sensitive strings that clears memory before deletion
-static void SensitiveStringDeleter(std::string* str) {
-    if (str && !str->empty()) {
-        OPENSSL_cleanse(&(*str)[0], str->size());
-        str->clear();
-    }
-    delete str;
-}
-
-BSSL_NAMESPACE_BEGIN
-BORINGSSL_MAKE_DELETER(std::string, SensitiveStringDeleter)
-BSSL_NAMESPACE_END
+// The SensitiveStringDeleter is now defined in password.cc
```

Comment:
Comment is out of date, also no need to say where code used to be Git history will track all of that.